src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -255,31 +258,35 @@ internal TabItem AddTabItem(IViewExtension viewExtension, ContentControl content
         // This method triggers the close operation on the selected tab. 
         private void CloseTab(object sender, RoutedEventArgs e)
         {
-            string tabName = (sender as Button).CommandParameter.ToString();
-            CloseTab(tabName);
+            string tabName = (sender as Button).DataContext.ToString();
+
+            CloseExtension?.Invoke();
```

Comment:
I think we should pass the tabName.... can you check what type the tab's dataContext really is? is it actually the ViewExtension?

Reply:
Added tabname as an input to the event.